test/run.d

```diff
@@ -147,26 +151,53 @@ void verifyCompilerExists(string[string] env)
 Builds the binary of the tools required by the testsuite.
 Does nothing if the tools already exist and are newer than their source.
 */
-void ensureToolsExists(const TestTool[] tools ...)
+void ensureToolsExists(string[string] env, const TestTool[] tools ...)
 {
     resultsDir.mkdirRecurse;
 
     foreach (tool; tools.parallel(1))
     {
-        const targetBin = resultsDir.buildPath(tool).exeName;
-        const sourceFile = toolsDir.buildPath(tool ~ ".d");
+        string targetBin;
+        string sourceFile;
+        if (tool.linksWithTests)
+        {
+            targetBin = resultsDir.buildPath(tool).objName;
+            sourceFile = toolsDir.buildPath(tool, tool ~ ".d");
+        }
+        else
+        {
+            targetBin = resultsDir.buildPath(tool).exeName;
+            sourceFile = toolsDir.buildPath(tool ~ ".d");
+        }
         if (targetBin.timeLastModified.ifThrown(SysTime.init) >= sourceFile.timeLastModified)
             writefln("%s is already up-to-date", tool);
         else
         {
-            const command = [
-                hostDMD,
-                "-of"~targetBin,
-                sourceFile
-            ] ~ tool.extraArgs;
+            string[] command;
+            string[string]* commandEnv = null;
```

Comment:
What's the motivation for making this a pointer to a pointer?
A D hashmap is already a pointer.

Reply:
Ah, for some reason I thought it had value semantics, but looks like I was wrong.